cmake/modules/SwiftUtils.cmake

```diff
@@ -1,136 +1,174 @@
+# Module import guard
+if(DEFINED SWIFT_UTILS_MODULE_LOADED)
+  return()
+endif()
+set(SWIFT_UTILS_MODULE_LOADED TRUE)
+
+
 include(CMakeParseArguments)
 
+
 function(precondition var)
-  cmake_parse_arguments(
-    PRECONDITION # prefix
-    "NEGATE" # options
-    "MESSAGE" # single-value args
-    "" # multi-value args
+  set(options NEGATE)
+  set(single_value_args MESSAGE)
+  set(multi_value_args)
+
+  cmake_parse_arguments(PRE
+    "${options}"
+    "${single_value_args}"
+    "${multi_value_args}"
     ${ARGN})
 
-  if (PRECONDITION_NEGATE)
-    if (${var})
-      if (PRECONDITION_MESSAGE)
-        message(FATAL_ERROR "Error! ${PRECONDITION_MESSAGE}")
+  if(PRE_NEGATE)
+    if(${var})
+      if(PRE_MESSAGE)
+        message(FATAL_ERROR "Error! ${PRE_MESSAGE}")
       else()
-        message(FATAL_ERROR "Error! Variable ${var} is true or not empty. The value of ${var} is ${${var}}.")
+        message(FATAL_ERROR
+          "Error! Variable ${var} is true or not empty. "
+          "The value of ${var} is ${${var}}.")
       endif()
     endif()
   else()
-    if (NOT ${var})
-      if (PRECONDITION_MESSAGE)
-        message(FATAL_ERROR "Error! ${PRECONDITION_MESSAGE}")
+    if(NOT ${var})
+      if(PRE_MESSAGE)
+        message(FATAL_ERROR "Error! ${PRE_MESSAGE}")
       else()
-        message(FATAL_ERROR "Error! Variable ${var} is false, empty or not set.")
+        message(FATAL_ERROR
+          "Error! Variable ${var} is false, empty or not set.")
       endif()
     endif()
   endif()
 endfunction()
 
+
 # Assert is 'NOT ${LHS} ${OP} ${RHS}' is true.
 function(precondition_binary_op OP LHS RHS)
-  cmake_parse_arguments(
-    PRECONDITIONBINOP # prefix
-    "NEGATE" # options
-    "MESSAGE" # single-value args
-    "" # multi-value args
+  set(options NEGATE)
+  set(single_value_args MESSAGE)
+  set(multi_value_args)
+
+  cmake_parse_arguments(PRE_BIN_OP
+    "${options}"
+    "${single_value_args}"
+    "${multi_value_args}"
     ${ARGN})
 
-  if (PRECONDITIONBINOP_NEGATE)
-    if (${LHS} ${OP} ${RHS})
-      if (PRECONDITIONBINOP_MESSAGE)
-        message(FATAL_ERROR "Error! ${PRECONDITIONBINOP_MESSAGE}")
+  if(PRE_BIN_OP_NEGATE)
+    if(${LHS} ${OP} ${RHS})
+      if(PRE_BIN_OP_MESSAGE)
+        message(FATAL_ERROR "Error! ${PRE_BIN_OP_MESSAGE}")
       else()
         message(FATAL_ERROR "Error! ${LHS} ${OP} ${RHS} is true!")
       endif()
     endif()
   else()
```

Comment:
Same here. If we are changing this, lets use a return() to eliminate indentation.